smtp-transport/src/test/kotlin/no/nav/emottak/SmtpTransportIntegrationTest.kt

```diff
@@ -0,0 +1,135 @@
+package no.nav.emottak
+
+import com.nimbusds.jwt.SignedJWT
+import com.zaxxer.hikari.HikariDataSource
+import io.ktor.client.call.body
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.request.get
+import io.ktor.client.request.headers
+import io.ktor.client.statement.HttpResponse
+import io.ktor.client.statement.bodyAsText
+import io.ktor.http.HttpHeaders
+import io.ktor.http.HttpStatusCode
+import io.ktor.serialization.kotlinx.json.json
+import io.ktor.server.testing.ApplicationTestBuilder
+import io.ktor.server.testing.testApplication
+import io.micrometer.prometheus.PrometheusConfig
+import io.micrometer.prometheus.PrometheusMeterRegistry
+import no.nav.emottak.model.Payload
+import no.nav.emottak.repository.PayloadRepository
+import no.nav.security.mock.oauth2.MockOAuth2Server
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.BeforeAll
+import org.testcontainers.containers.PostgreSQLContainer
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotEquals
+
+class SmtpTransportIntegrationTest {
```

Comment:
Jeg er lit usikker på om du trenger en full blown integrasjonstest her. Jeg har allerede testet persistens-laget relativt grundig gjennom PayloadRepositorySpec.

Hadde også gjerne sett at vi beholder samme type teststruktur. Fint om du mapper om denne til Kotest-standarden og bruker StringSpec. Det blir litt rotete med to forskjellige test-rammeverk i samme prosjekt. 

Jeg antar du har tatt litt inspirasjon av hva som allerede finnes i send-in-send-out, men smtp-modulen er strengt tatt en helt selvstendig, avgrenset modul med dedikert funksjonalitet. Planen er å flytte denne ut av send-in-send-out.